--- mips.H	2013-04-23 10:46:52.000000000 -0300
+++ mips_acSynth.H	2013-05-01 16:02:41.226331000 -0300
@@ -23,7 +23,9 @@
 #include "mips_arch.H"
 #include "mips_isa.H"
 #include "mips_syscall.H"
-
+#ifdef POWER_SIM
+#include "arch_power_stats.H"
+#endif
 
 class mips: public ac_module, public mips_arch {
 private:
@@ -32,7 +34,10 @@
 public:
 
   unsigned bhv_pc;
- 
+#ifdef POWER_SIM
+  power_stats ps;
+#endif
+
    /****/
    sc_core::sc_time time_info;
    sc_core::sc_time getTimeInfo() { return time_info;} ;
@@ -58,7 +63,11 @@
   SC_HAS_PROCESS( mips );
 
   //!Constructor.
+#ifdef POWER_SIM
+  mips( sc_module_name name_ ): ac_module(name_), mips_arch(), ISA(*this), ps((const char*)name_), time_info(0,SC_NS) {
+#else
   mips( sc_module_name name_ ): ac_module(name_), mips_arch(), ISA(*this), time_info(0,SC_NS) {
+#endif
 
     SC_THREAD( behavior );
     bhv_pc = 0; 
