--- sparc.cpp	2013-04-28 20:38:29.000000000 -0300
+++ sparc_acSynth.cpp	2013-05-01 16:48:24.602282758 -0300
@@ -568,7 +568,12 @@
       } // switch (ins_id)
       break;
     }
-    if ((!ac_wait_sig) && (!ac_annul_sig)) ac_instr_counter+=1;
+    if ((!ac_wait_sig) && (!ac_annul_sig)) {
+      ac_instr_counter+=1;
+#ifdef POWER_SIM
+      ps.update_stat_power(ins_id);
+#endif
+    }
     ac_annul_sig = 0;
   }
 //!Updating Regs for behavioral simulation.
